sgpt/utils.py

```diff
@@ -10,9 +10,16 @@
 
 
 class ModelOptions(str, Enum):
-    GPT3 = "gpt-3.5-turbo"
-    GPT4 = "gpt-4"
-    GPT4_32K = "gpt-4-32k"
+    """
+    Model endpoint compatibility
+    https://platform.openai.com/docs/models/model-endpoint-compatibility
+    """
+    GPT4 = 'gpt-4'
+    GPT40314 = 'gpt-4-0314'
+    GPT432k = 'gpt-4-32k'
+    GPT432k0314 = 'gpt-4-32k-0314'
+    GPT35TURBO = 'gpt-3.5-turbo'
+    GPT35TURBO0301 = 'gpt-3.5-turbo-0301'
```

Comment:
I have concerns about including models that will be phased out with each new version. The OpenAI documentation specifically states:

> This model will not receive updates and is scheduled for deprecation 3 months post the release of a new version.

It may not be efficient to utilize such models, given their short lifespan and lack of continued support.

Reply:
I think this is a valid concern -- it feels like needless maintenance to constantly be updating a dataclass. However, without the update it limits user's ability to choose the model they are using. With [today's announcement](https://openai.com/blog/function-calling-and-other-api-updates) this includes a range of model options...

### I see three options:
1. only allow for the default model option: by only updating `ModelOptions` with non-date models that wont be deprecated
2. update with all available options: (as above) feels like needless maintenance
3. remove the data class: least restrictive but will return ambiguous error when a non-compatible model is used

**Personally I think 2 is low-effort with respect to PR review time, and adds zero complexity to the control flow**

Your call!

Reply:
*Updated PR with all options*

Reply:
I personally prefer the first option. However, for the time being, let's proceed with merging this. I'll ensure to remove any non-date models prior to releasing the next version of sgpt.